node/src/components/gossiper/provider_impls/block_provider.rs

```diff
@@ -1,24 +1,24 @@
+use async_trait::async_trait;
+
 use crate::{
-    components::gossiper::{Event, GossipItem, Gossiper, ItemProvider},
-    effect::{requests::StorageRequest, EffectBuilder, EffectExt, Effects},
-    types::{Block, BlockHash, NodeId},
+    components::gossiper::{GossipItem, Gossiper, ItemProvider},
+    effect::{requests::StorageRequest, EffectBuilder},
+    types::{Block, BlockHash},
 };
 
+#[async_trait]
```

Comment:
I'm not against it, but I'm curious why the move towards async functions here. It's breaking the pattern of using and returning `Effects`.

Reply:
Initially it was because I thought we'd have a couple of different places where we'd be calling `is_stored` with a different flow for each after the result is yielded.  Having a specific event chained to the end of the trait method made that clunky - I was looking at adding a third trait method which effectively also checked for the data in storage but went to different event on returning.

Turns out that wasn't actually needed, but I didn't change this back since it avoids duplication of the `.event(move |result| { Event::IsStoredResult { ... }})` in each of the provider impls.